wgpu/src/lib.rs

```diff
@@ -4739,7 +4767,42 @@ impl<'a> Drop for RenderPass<'a> {
     }
 }
 
-impl ComputePass {
+impl<'encoder> ComputePass<'encoder> {
+    /// Drops the lifetime relationship to the parent command encoder, making usage of
+    /// the encoder while this pass is recorded a run-time error instead.
+    ///
+    /// Attention: As long as the compute pass has not been ended, any mutating operation on the parent
+    /// command encoder will cause a runtime error and invalidate it!
+    /// By default the lifetime constraint protects from this, but it can be useful
+    /// to handle this at run time, for example in order to store the pass and encoder in the same
+    /// data structure.
+    ///
+    /// This operation has no effect on pass recording and is considered low overhead.
+    /// It's a safe operation, since [`CommandEncoder`] is in a locked state as long as the pass is active
+    /// regardless of here lifted lifetime constraint.
+    pub fn make_static(mut self) -> ComputePass<'static> {
+        // It's tempting to simply transmute the object, but that would be unsound
+        // since `repr(rust)` objects have no guarantees about padding or alignment
```

Comment:
Nitpick: I'm pretty sure it's impossible for lifetimes to make a difference in layout, because it _must_ be possible to pass a `&Foo<'static>` to a `for<'a, 'b> fn(&'a Foo<'b>)` — and there's no opportunity for the compiler to monomorphize anything there, so the layouts involved must be identical. I don't know that this has been officially written down anywhere, though.

However, I'm not suggesting that this *should* be written in the unsafe fashion; rather, a better reason not to do this is simply “avoid gratuitous unsafe code”. (But before spending any time on a new comment, see the below comment about a different implementation.)